subsys/usb/class/bluetooth.c

```diff
@@ -168,14 +215,43 @@ static void hci_rx_thread(void)
 	}
 }
 
-static void hci_tx_thread(void)
+static void usb_h4_recv(struct net_buf *buf)
+{
+	u8_t type;
+
+	type = net_buf_pull_u8(buf);
+
+	switch (type) {
+	case H4_CMD:
+		bt_buf_set_type(buf, BT_BUF_CMD);
+		break;
+	case H4_ACL:
+		bt_buf_set_type(buf, BT_BUF_ACL_OUT);
+		break;
+	default:
+		LOG_ERR("Unknown H4 type %u", type);
+		return;
+	}
+
+	LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+
+	if (bt_send(buf)) {
+		LOG_ERR("Error sending to driver");
+		net_buf_unref(buf);
+	}
+}
```

Comment:
This is similar to ```usb_h4_send```. I prefer one of the solitions:
- rename it to something like ```h4_cmd_type_update```. This is what actually happens there. And leave ```bt_send``` untouched in ```hci_rx_thread```
```diff
diff --git a/subsys/usb/class/bluetooth.c b/subsys/usb/class/bluetooth.c
index 81fe773834..f1e1bdd64a 100644
--- a/subsys/usb/class/bluetooth.c
+++ b/subsys/usb/class/bluetooth.c
@@ -215,7 +215,7 @@ static void hci_tx_thread(void)
 	}
 }
 
-static void usb_h4_recv(struct net_buf *buf)
+static void usb_h4_cmd_type_update(struct net_buf *buf)
 {
 	u8_t type;
 
@@ -234,11 +234,6 @@ static void usb_h4_recv(struct net_buf *buf)
 	}
 
 	LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
-
-	if (bt_send(buf)) {
-		LOG_ERR("Error sending to driver");
-		net_buf_unref(buf);
-	}
 }
 
 static void hci_rx_thread(void)
@@ -250,8 +245,10 @@ static void hci_rx_thread(void)
 
 		if (IS_ENABLED(CONFIG_USB_DEVICE_BLUETOOTH_VS_H4) &&
 		    mode == BT_HCI_VS_USB_H4_MODE) {
-			usb_h4_recv(buf);
-		} else if (bt_send(buf)) {
+			usb_h4_cmd_type_update(buf);
+		}
+
+		if (bt_send(buf)) {
 			LOG_ERR("Error sending to driver");
 			net_buf_unref(buf);
 		}
```
- put this code into ```hci_rx_thread``` directly. It only changes the command type and later calls ```bt_send``` exactly the same way the H2 does.
``` diff
diff --git a/subsys/usb/class/bluetooth.c b/subsys/usb/class/bluetooth.c
index 81fe773834..e31264bc0d 100644
--- a/subsys/usb/class/bluetooth.c
+++ b/subsys/usb/class/bluetooth.c
@@ -215,32 +215,6 @@ static void hci_tx_thread(void)
 	}
 }
 
-static void usb_h4_recv(struct net_buf *buf)
-{
-	u8_t type;
-
-	type = net_buf_pull_u8(buf);
-
-	switch (type) {
-	case H4_CMD:
-		bt_buf_set_type(buf, BT_BUF_CMD);
-		break;
-	case H4_ACL:
-		bt_buf_set_type(buf, BT_BUF_ACL_OUT);
-		break;
-	default:
-		LOG_ERR("Unknown H4 type %u", type);
-		return;
-	}
-
-	LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
-
-	if (bt_send(buf)) {
-		LOG_ERR("Error sending to driver");
-		net_buf_unref(buf);
-	}
-}
-
 static void hci_rx_thread(void)
 {
 	while (true) {
@@ -250,8 +224,22 @@ static void hci_rx_thread(void)
 
 		if (IS_ENABLED(CONFIG_USB_DEVICE_BLUETOOTH_VS_H4) &&
 		    mode == BT_HCI_VS_USB_H4_MODE) {
-			usb_h4_recv(buf);
-		} else if (bt_send(buf)) {
+			switch (net_buf_pull_u8(buf)) {
+			case H4_CMD:
+				bt_buf_set_type(buf, BT_BUF_CMD);
+				break;
+			case H4_ACL:
+				bt_buf_set_type(buf, BT_BUF_ACL_OUT);
+				break;
+			default:
+				LOG_ERR("Unknown H4 type %u", type);
+				return;
+			}
+		}
+
+		LOG_DBG("buf %p type %u len %u", buf, bt_buf_get_type(buf), buf->len);
+
+		if (bt_send(buf)) {
 			LOG_ERR("Error sending to driver");
 			net_buf_unref(buf);
 		}
```


Same comment applies for ```usb_h4_recv``` and ```hci_rx_thread``` in bt_h4.c

Reply:
Right, so when I have to replace usb_h4_recv I would have to remove these all over the place, no thanks.